libs/pika/mpi_base/src/mpi_exception.cpp

```diff
@@ -1,19 +1,20 @@
+//  Copyright (c) 2024 ETH Zurich
 //  Copyright (c) 2020 John Biddiscombe
 //
 //  SPDX-License-Identifier: BSL-1.0
 //  Distributed under the Boost Software License, Version 1.0. (See accompanying
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 #include <pika/config.hpp>
-#include <pika/async_mpi/mpi_exception.hpp>
 #include <pika/errors/exception.hpp>
-#include <pika/mpi_base/mpi.hpp>
+#include <pika/mpi_base/mpi_exception.hpp>
 
 #include <cstddef>
 #include <memory>
+#include <mpi.h>
```

Comment:
Please include `pika/mpi_base/mpi.hpp` instead. It suppresses warnings from `mpi.h` that we'd otherwise need to suppress everywhere that `mpi.h` is included.